include/adc.h

```diff
@@ -26,121 +26,324 @@ extern "C" {
+	ADC_GAIN_1_3, /**< x 1/3. */
+	ADC_GAIN_1_2, /**< x 1/2. */
+	ADC_GAIN_2_3, /**< x 2/3. */
+	ADC_GAIN_1,   /**< x 1. */
+	ADC_GAIN_2,   /**< x 2. */
+	ADC_GAIN_3,   /**< x 3. */
+	ADC_GAIN_4,   /**< x 4. */
+	ADC_GAIN_8,   /**< x 8. */
+	ADC_GAIN_16,  /**< x 16. */
+	ADC_GAIN_32,  /**< x 32. */
+	ADC_GAIN_64,  /**< x 64. */
+};
+
+/** @brief ADC references. */
+enum adc_reference {
+	ADC_REF_VDD_1,     /**< VDD. */
+	ADC_REF_VDD_1_2,   /**< VDD/2. */
+	ADC_REF_VDD_1_3,   /**< VDD/3. */
+	ADC_REF_VDD_1_4,   /**< VDD/4. */
+	ADC_REF_INTERNAL,  /**< Internal. */
+	ADC_REF_EXTERNAL0, /**< External, input 0. */
+	ADC_REF_EXTERNAL1, /**< External, input 1. */
+};
+
+/** Acquisition time is expressed in microseconds. */
+#define ADC_ACQ_TIME_MICROSECONDS  (1u)
+/** Acquisition time is expressed in nanoseconds. */
+#define ADC_ACQ_TIME_NANOSECONDS   (2u)
+/** Acquisition time is expressed in ADC ticks. */
+#define ADC_ACQ_TIME_TICKS         (3u)
+/** Macro for composing the acquisition time value in given units. */
+#define ADC_ACQ_TIME(unit, value)  (((unit) << 14) | ((value) & BIT_MASK(14)))
+/** Value indicating that the default acquisition time should be used. */
+#define ADC_ACQ_TIME_DEFAULT       0
+
+#define ADC_ACQ_TIME_UNIT(time)    (((time) >> 14) & BIT_MASK(2))
+#define ADC_ACQ_TIME_VALUE(time)   ((time) & BIT_MASK(14))
+
 /**
- * @brief ADC driver Sequence entry
- *
- * This structure defines a sequence entry used
- * to define a sample from a specific channel.
+ * @brief Structure for specifying the configuration of an ADC channel.
  */
-struct adc_seq_entry {
-	/** Clock ticks delay before sampling the ADC. */
-	s32_t sampling_delay;
+struct adc_channel_cfg {
+	/** Gain selection. */
+	enum adc_gain gain;
+
+	/** Reference selection. */
+	enum adc_reference reference;
+
+	/**
+	 * Acquisition time.
+	 * Use the ADC_ACQ_TIME macro to compose the value for this field or
+	 * pass ADC_ACQ_TIME_DEFAULT to use the default setting for a given
+	 * hardware (e.g. when the hardware does not allow to configure the
+	 * acquisition time).
+	 * Particular drivers do not necessarily support all the possible units.
+	 * Value range is 0-16383 for a given unit.
+	 */
+	u16_t acquisition_time;
+
+	/**
+	 * Channel identifier (0-31).
+	 * This value identifies the channel only within the ADC API - when a
+	 * read request is done, the corresponding bit in the "channels" field
+	 * of the "adc_sequence" structure must be set to include this channel
+	 * in the sampling. In particular, this identifier does not directly
+	 * select any physical input of the ADC. The ADC input to be associated
+	 * with the channel must be indicated in the "input_positive" field
+	 * (and for differential channels the "input_negative" field must be
+	 * also used to compose a pair of inputs).
+	 */
+	u8_t channel_id   : 5;
+
+	/** Channel type: single-ended or differential. */
+	u8_t differential : 1;
+
+	/** Positive ADC input. */
+	u8_t input_positive;
```

Comment:
What user populate in this variable ? pin number from which positive input should taken ?